include/mbedtls/debug.h

```diff
@@ -131,6 +131,10 @@
 #endif \
     /* (defined(__MINGW32__)  && __USE_MINGW_ANSI_STDIO == 0) || (defined(_MSC_VER) && _MSC_VER < 1800) */
 
+#if !defined(MBEDTLS_PRINTF_MS_TIME)
+#define MBEDTLS_PRINTF_MS_TIME PRId64
```

Comment:
(comment) it really feels like we should be able to set this based on `sizeof(mbedtls_ms_time_t)`, but I don't think you can use `sizeof()` in the pre-processor according to the standard (might be an extension on some compilers)

Reply:
The preprocessor operates before type definitions are proecessed, so even as an extension, `sizeof` couldn't work reliably with e.g. a `typedef`'ed name.

C is generally incapable of abstracting over a type. C99 specifies predefined macros like `PRIu64` but that only helps if you know that a type is `uintNN_t` and you have NN as a token, not if all you have is a token that's a type name. The standard solution (since C99) is to cast to `uintmax_t` or `intmax_t` (you have to know the type's signedness) and use `PRIuMAX` or `PRIdMAX` to print. But that assumes that the libc supports `PRIuMAX` and I'm not confident that in practice all embedded libc that implement `printf` do support `PRIuMAX` (for example, its integer printing might be limited to 32-bit or 64-bit integers, even if the compiler supports 64-bit or 128-bit integers).